backbone/src/main/java/org/researchstack/backbone/ui/ConsentTaskActivity.java

```diff
@@ -80,7 +89,51 @@ public void onSaveStep(int action, Step step, StepResult result) {
             consentHtml = ((ConsentDocumentStep) step).getConsentHTML();
         }
 
-        super.onSaveStep(action, step, result);
+        StepCallbacks stepCallbacks = getCurrentFragmentStepCallbacks();
```

Comment:
You should use the viewModel instead of getting a reference to the fragment that is on top. 
The viewModel is shared between the activity and the fragment.

if you need to fire the event